docs/architecture/adr-034-change-pubkey.md

```diff
@@ -0,0 +1,63 @@
+# ADR 034: Change PubKey
```

Comment:
Please rename the title to something more specific. I propose `Account Rekeying`. Or add some context, eg: `Change PubKey Msg`, `Change PbKey  service` ...

I think Algorand originally introduced this feature and coined the term [Rekeying](https://www.algorand.com/resources/blog/announcing-rekeying)